StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1284,7 +1284,7 @@ public class STPPaymentHandler: NSObject {
 
             if let mobileAuthURL = authenticationAction.cashAppRedirectToApp?.mobileAuthURL {
                 let resultingRedirectURL = self.followRedirects(to: mobileAuthURL, urlSession: URLSession.shared)
-                _handleRedirect(to: resultingRedirectURL, withReturn: returnURL)
+                _handleRedirectToExternalBrowser(to: resultingRedirectURL, withReturn: returnURL)
 
             } else {
```

Comment:
Do we really want to use `_handleRedirectToExternalBrowser(to: resultingRedirectURL, withReturn: returnURL)` ?

If the app is not installed or safari isn't set as their default browser, then the user will end up in web browser outside of the merchant app.

Reply:
I actually have Firefox set as my default in the video above, since we are opening a universal link the external browser is just skipped and it goes straight to Cash App! If the app isn't installed it goes to the app store. The behavior between SFSafariViewController and the external flow is actually the same, but the callbacks for the external browser are a little more favorable for a fast redirect PM from my testing.

Reply:
Ah, that's interesting -- probably because it is not a universal link... in this case, I still don't think we should be using redirect to external browser, we should probably do:
```
            if let mobileAuthURL = authenticationAction.cashAppRedirectToApp?.mobileAuthURL {
                let resultingRedirectURL = self.followRedirects(to: mobileAuthURL, urlSession: URLSession.shared)
                _handleRedirect(to: resultingRedirectURL, fallbackURL: mobileAuthURL, return: returnURL)
             }

```
This path seems to use `UIApplication.shared.open(` and assume the link is a customURL.  If it succeeds, then great, the app should open.  If this fails, because the app is not installed, open up the original value of `authenticationAction.cashAppRedirectToApp?.mobileAuthURL` in an in-app web modal.

Reply:
Swapped, also just saw the `_handleRedirect` function throws the `requiredAppNotAvailable` error if it can't open the link :)